pkg/config/structure/unmarshal.go

```diff
@@ -38,10 +39,21 @@ var ConvertEmptyStringToNil UnmarshalKeyOption = func(fs *featureSet) {
 }
 
 // UnmarshalKey retrieves data from the config at the given key and deserializes it
-// to be stored on the target struct. It is implemented entirely using reflection, and
-// does not depend upon details of the data model of the config.
-// Target struct can use of struct tag of "yaml", "json", or "mapstructure" to rename fields
+// to be stored on the target struct.
+//
+// When DD_CONF_NODETREEMODEL is enabled we use the implementation using reflection, and do not depend upon details of
+// the data model of the config. Target struct can use of struct tag of "yaml", "json", or "mapstructure" to rename fields
+//
+// Else the viper/legacy version is used.
 func UnmarshalKey(cfg model.Reader, key string, target interface{}, opts ...UnmarshalKeyOption) error {
+	nodetreemodel := os.Getenv("DD_CONF_NODETREEMODEL")
+	if nodetreemodel == "enabled" {
```

Comment:
The check in `pkg/config/setup/config.go` is using "enable" instead, just one letter off. I also think it would be useful to be able to enable this without enabling the entire nodetreemodel for the config.

So instead, how about:
```suggestion
	if nodetreemodel == "enable" || nodetreemodel == "unmarshal" {
```

and then also document this value "unmarshal" [here](https://github.com/DataDog/datadog-agent/pull/30622/files#diff-a7b6619528f155e64cefe12cc4ba753eb7688c87fdf9d96e1052a12eeb519df5L248)